### ISSUE TITLE
Fix view info select statement for local indexes

### ISSUE THREAD

**Issue by psarna** [scylladb/scylla#5241]
Calculating the select statement for given view_info structure
used to work fine, but once local indexes were introduced, a subtle
bug appeared: the legacy token column does not exist in local indexes
and a valid clustering key column was omitted instead.
That results in potentially incorrect partition slices being used later
in read-before-write.
There's a long term plan for removing select_statement from
view info altogether, but nonetheless the bug needs to be fixed first.

This issue is already solved by ff45879ce2215a2b6644cb01b79b4fb49ed137cf, but is retroactively added for backporting purposes (3.1).

**Comment by nyh** [comment]
So, as Piotr said, this issue was already solved on the master branch, so there is no need for it to remain open. Backporting patches can still refer to closed issues.

**Comment by tzach** [comment]
@psarna can you please provide a user-level (CQL) example of this issue?
Is it a correctness or performance issue?

**Comment by psarna** [comment]
It's a little bit of both. In order to see the effects you'd have to:
1. Create a local index on a cluster that does not support computed columns yet
2. (correctness) Describe the underlying materialized view of this index - it will not contain a correct description. It's meaningless as this materialized view is created correctly (and should never be created directly), but the fact remains that its description does not match what's actually inside it
3. (performance) When calculating if view updates must be generated for an insertion, one of the things that happen is that `affected clustering ranges` are computed and used to decide if these ranges affect views. With this bug, these ranges will sometimes be too broad, which means that we'd sometimes start trying to generate mutations for views and ending up with no changes generated and some CPU wasted.

To sum up, neither correctness nor performance are really problematic, it's a minor issue.
